src/bio/bio_xstream.c

```diff
@@ -108,6 +108,12 @@ bio_spdk_env_init(void)
 
 	opts.env_context = (char *)dpdk_cli_override_opts;
 
+	rc = bio_set_hotplug_filter(nvme_glb.bd_nvme_conf);
```

Comment:
I think Silong is pushing a patch to initialize spdk (for using hugepages) even without bdev configured, so you'd skip set hotplug filter when bd_nvme_conf isn't specified to avoid potential merge conflicts.